crates/staged-sync/src/utils/init.rs

```diff
@@ -62,7 +67,12 @@ pub fn init_genesis<DB: Database>(
     let tx = db.tx_mut()?;
     insert_genesis_state::<DB>(&tx, genesis)?;
 
+    // use transaction to insert genesis header
+    let transaction = Transaction::new_raw(&db, tx);
+    insert_genesis_hashes(transaction, genesis)?;
+
     // Insert header
+    let tx = db.tx_mut()?;
```

Comment:
this should fail to acquire the write transaction because it's already owned by `Transaction`